ipld/README.md

```diff
@@ -23,11 +25,45 @@ A _merkle-link_ is a link between two objects which is content-addressed with th
 - **Cryptographic Integrity Checking**: resolving a link's value can be tested by hashing. In turn, this allows wide, secure, trustless exchanges of data (e.g. git or bittorrent), as others cannot give you any data that does not hash to the link's value.
 - **Immutable Datastructures**: data structures with merkle links cannot mutate, which is a nice property for distributed systems. This is useful for versioning, for representing distributed mutable state (eg CRDTs), and for long term archival.
 
-A _merkle-link_ is represented in the IPLD object model by a map containing a key `@link` mapped to a string value: the actual link. When dereferencing the link, the map itself is to be replaced by the object it points to unless the link path is invalid.
+A _merkle-link_ is represented in the IPLD object model by a map containing a single key `/` mapped to a "link value". For example:
+
+
+**A link, represented in json as a "link object"**
+
+```js
+{ "/" : "/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k" }
+// "/" is the link key
+// "/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k" is the link value
+```
+
+**Object with a link at `foo/baz`**
 
-The link can either be a base58 hash, in which case it is assumed that it is a link in the `/ipfs` hierarchy, or directly the absolute path to the object. Currently, only the `/ipfs` hierarchy is allowed.
+```js
+{
+  "foo": {
+    "bar": "/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k", // not a link
+    "baz": {"/": "/ipfs/QmUmg7BZC1YP1ca66rRtWKxpXp77WgVHrnv263JtDuvs2k"} // link
+  }
+}
 
-If an application wants to use the `@link` key for other purposes, the application itself is responsible to escape the keys in the IPLD object so that the application keys do not conflict with the `@link` key. When discussing application specific paths, it may be worth escaping all keys starting with `@` in case future versions of IPLD make use of other keys.
+**Object with pseudo "link object" at `files/cat.jpg` and actual link at `files/cat.jpg/link`**
```

Comment:
This seems to be new to me and problematic when building it into the resolver. I was under the impression key like `"link"` would not have a special meaning, but with this you are effectively giving them one. 
Also this seems ambiguous to me, consider this object:

```js
{
  "files: {
    "cat.jpg": {
      "coolLink": {"/": "/ipfs/Qm..AAA"},
      "link": {"/": "/ipfs/QmBBB"},
      "mode": 0755
    }
  }
}
```

then how should `files/cat.jpg` be resolved? Objects are not inherently ordered in JavaScript so we can not guarantee the same resolution if we say first come, first serve.